### PR TITLE
Allow to specify Classloader

### PR DESCRIPTION
**Pull request by Rovak**
### Motivation

When using Play Framework there appear to be classloader errors when deserializing case classes which reference other classes in their parameter types.

### Example

It works in play dev mode for the simple cases, i narrowed the error down to the following, this works:

```scala
case class Person(name: String)

val person = genson.fromJson[Person]("{\"name\":\"John Doe\"}")
```

When the case class references another class then it will thrown an error that the class Person was not found:

```scala
case class Person(name: String)
case class PhoneBook(names: List[Person])

val phonebook = genson.fromJson[PhoneBook]("{\"persons\":[{\"name\":\"John Doe\"}]}")
```

## TODO

- [x] Add `withClassLoader` to GensonBuilder
- [x] Check where other classloaders are used and if they should use the classloader specified in the GensonBuilder

### CODE REVIEW

**Comment by EugenCepoi** [comment]
Let me know when you have added the useClassloader method/it is ready to merge

**Comment by Rovak** [comment]
@EugenCepoi i've added the `withClassLoader`

File: genson/src/main/java/com/owlike/genson/GensonBuilder.java [hunk 1]
```diff
@@ -60,6 +60,7 @@
   private VisibilityFilter methodFilter = VisibilityFilter.PACKAGE_PUBLIC;
   private VisibilityFilter constructorFilter = VisibilityFilter.PACKAGE_PUBLIC;
 
+  private ClassLoader classLoader = null;
```
Comment (EugenCepoi) [inline genson/src/main/java/com/owlike/genson/GensonBuilder.java]: Why not initialize it to `GensonBuilder.getClassLoader` and then just do new `ScalaBeanPropertyFactory(builder.getClassLoader())`? An additional not null check could be added to withClassLoader method.

Comment (Rovak) [inline genson/src/main/java/com/owlike/genson/GensonBuilder.java]: something like this?

```java
public ClassLoader getClassLoader() {
  if (classLoader == null)
    classLoader = this.getClass().getClassLoader();

  return classLoader;
}
```

Comment (EugenCepoi) [inline genson/src/main/java/com/owlike/genson/GensonBuilder.java]: Why not, but I was thinking directly where it is initialized to null now.

Comment (Rovak) [inline genson/src/main/java/com/owlike/genson/GensonBuilder.java]: ah right, changed it as you suggested

**Comment by EugenCepoi** [comment]
Looks good, thanks!

**Pull request merged**